chainermn/communicators/_base.py

```diff
@@ -15,6 +19,19 @@ def rank(self):
     def size(self):
         return self.mpi_comm.size
 
+    def send(self, array, dest, tag):
+        shape = numpy.array(array.shape, dtype='int')
```

Comment:
If there is no good reason, how about using `np.int32` instead of using type strings? We are consistently using dtype objects (e.g., `np.int32`) in ChainerMN. FYI: `cupy.int32 == numpy.int32`